README.md

```diff
@@ -4,9 +4,146 @@ CartoDB Spatial Analysis extension for PostgreSQL.
+* - *src/py* Python module source code
+* *release* reseleased versions
+* *env* base directory for Python virtual environments
 
 ## Requirements
 
-* pip
+* pip, virtualenv, PostgreSQL
+* python-scipy system package (see src/py/README.md)
+
+# Working Process
+
+We use the default `develop` branch as the basis for development.
+This branch and `master` are maintained by the *Release Manager*.
+The `master` branch is used to merge and tag releases to be
+deployed in production.
+
+In addition to these two permanent branches, temporal *topic*
+branches will be used for all modifications.
+
+## Development
+
+A topic branch should be created out of the `develop` branch
+and be used for the development process; see src/py/README.md
+
+Modifications are done inside `src/pg/sql` and `src/py/crankshaft`.
+Take into account:
+
+*  Always remember to add tests (`src/pg/test`, `src/py/crankshaft/test`)
+   for any new functionality.
+*  Add or modify the corresponding documentation files in the `doc` folder.
+   Since we expect to have highly technical functions here, an extense
+   background explanation would be of great help to users of this extension.
+*  Convention: Use snake case (i.e. `snake_case` and not `CamelCase`) for all
+   functions. Prefix functions intended for public use with `cdb_`
+   and private functions (to be used only internally inside
+   the extension)  with `_cdb_`.
+
+Update the local development installation with `sudo make install`.
+This will update the 'dev' version of the extension in 'src/pg/' and
+make it available to PostgreSQL.
+It will also install the python package (crankshaft) in a virtual
+environment `env/dev`.
+
+Run the tests with `make test`
+
+To use the python extension for custom tests, activate the virtual
+environment with:
+
+```
+source envs/dev/bin/activate
+```
+
+Update extension in a working database with:
+
+* `ALTER EXTENSION crankshaft VERSION TO 'current';`
+  `ALTER EXTENSION crankshaft VERSION TO 'dev';`
+
+Note: we keep the current development version install as 'dev' always;
+we update through the 'current' alias to allow changing the extension
+contents but not the version identifier. This will fail if the
+changes involve incompatible function changes such as a different
+return type; in that case the offending function (or the whole extension)
+should be dropped manually before the update.
+
+If the extension has not previously been installed in a database
+we can:
+
+* `CREATE EXTENSION crankshaft WITH VERSION 'dev';`
+
+Note: the development extension uses the development pyhton virtual
+environment automatically.
+
+Once the tests are succeeding a new Pull-Request can be created.
+CI-tests must be checked to be successful.
+
+Before proceeding to the release process peer code reviewing of the code is a must.
+
+## Release
+
+The release process of a new version of the extension
+shall by performed by the designated *Release Manager*.
```

Comment:
the release part can probably be moved to a separate doc, but ok keeping it all together here FTM